src/battle/partner/parakarry.c

```diff
@@ -1,7 +1,336 @@
+            D_8023BD98[0] = hudID;
+            hud_element_set_render_depth(hudID, 9);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimShimmerE);
+            D_8023BD98[1] = hudID;
+            hud_element_set_render_depth(hudID, 9);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimShimmerD);
+            D_8023BD98[2] = hudID;
+            hud_element_set_render_depth(hudID, 9);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimShimmerC);
+            D_8023BD98[3] = hudID;
+            hud_element_set_render_depth(hudID, 9);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimShimmerB);
+            D_8023BD98[4] = hudID;
+            hud_element_set_render_depth(hudID, 9);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimShimmerA);
+            D_8023BD98[5] = hudID;
+            hud_element_set_render_depth(hudID, 9);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+
+            hudID = hud_element_create(HudScript_AimTarget);
+            D_8023BDB4 = hudID;
+            hud_element_set_render_depth(hudID, 0xA);
+            hud_element_create_transform_A(D_8023BDB4);
+            hud_element_set_render_pos(hudID, -0x64, -0x64);
+            D_8023BDC4 = -0x30;
+            D_8023BDC8 = 0x50;
+
+            hudID = hud_element_create(HudScript_StickHoldLeft);
+            D_8023BDC0 = hudID;
+            hud_element_set_render_pos(hudID, D_8023BDC4, D_8023BDC8);
+            hud_element_set_render_depth(hudID, 0);
+
+            set_goal_pos_to_part(state, parakarry->targetActorID, parakarry->targetPartIndex);
+            targetActor = get_actor(parakarry->targetActorID);
+            targetActorPart = get_actor_part(targetActor, parakarry->targetPartIndex);
+
+            state->goalPos.x += targetActorPart->unk_75 * targetActor->scalingFactor;
+            state->goalPos.y += targetActorPart->unk_76 * targetActor->scalingFactor;
+            state->goalPos.z = state->goalPos.z;
+            state->unk_24 = (targetActorPart->size.y + targetActorPart->size.x) / 2 / 24.0;
+            hud_element_set_scale(D_8023BDB4, state->unk_24 * targetActor->scalingFactor);
+            state->currentPos.x = parakarry->currentPos.x + 8.0f;
+            state->currentPos.y = parakarry->currentPos.y + 16.0f;
+            state->currentPos.z = parakarry->currentPos.z;
+            state->angle = atan2(state->currentPos.x, state->currentPos.y, state->goalPos.x, state->goalPos.y);
+            state->bounceDivisor = state->angle;
+            state->distance = 116.0f;
+            state->unk_18.x = state->angle;
+            i = 0;
+
+            do {
```

Comment:
Can this be a for loop with two exit conditions (i < 30, screenY < 181)? I feel like that'd be nicer (also applies below)